packages/components/src/internal/components/domainproperties/ScannableOption.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FC, memo, useCallback } from 'react';
+import { Col, FormControl, Row } from 'react-bootstrap';
+import { DomainFieldLabel } from './DomainFieldLabel';
+import { createFormInputId, createFormInputName } from './actions';
+import { DOMAIN_FIELD_SCANNABLE_OPTION } from './constants';
+import { isFieldFullyLocked } from './propertiesUtil';
+import { ITypeDependentProps } from './models';
+
+export interface ScannableProps extends ITypeDependentProps{
+    scannable?: boolean;
+    appPropertiesOnly?: boolean;
+    showScannableOption?: boolean;
+}
+
+export const ScannableOption: FC<ScannableProps> = memo(props => {
+    const { domainIndex, index, lockType, scannable = false, appPropertiesOnly, showScannableOption, onChange } = props;
+    if (!appPropertiesOnly || !showScannableOption) return undefined;
+
+    const handleOptionToggle = useCallback((event: any): void => {
+        const {id, checked} = event.target;
+        onChange?.(id, checked);
+    },[onChange]);
+
+    return (
+        <>
+            <Row>
+                <Col xs={3}>
+                    <div className="domain-field-label">
+                        <DomainFieldLabel
+                            label="Barcode Field"
+                            helpTipBody={
+                                'When using the Find Samples dialog from the search bar and choosing the\n' +
+                                '                                        "Barcodes" option, text fields\n' +
```

Comment:
should this text now say "text and numeric fields"?

Reply:
yes, updated.